lib/staking.ts

```diff
@@ -0,0 +1,25 @@
+import { QueryClient, StakingExtension, setupStakingExtension } from "@cosmjs/stargate";
+import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
+import { Validator } from "cosmjs-types/cosmos/staking/v1beta1/staking";
+
+const getValidatorsPage = (
+  queryClient: QueryClient & StakingExtension,
+  paginationKey: Uint8Array | undefined,
+) => queryClient.staking.validators("BOND_STATUS_BONDED", paginationKey);
+
+export const getAllValidators = async (rpcUrl: string): Promise<readonly Validator[]> => {
+  const validators: Validator[] = [];
+
+  const tmClient = await Tendermint34Client.connect(rpcUrl);
```

Comment:
Can you use `await connectComet(rpcUrl))`? This does auto-detection of the backend version